File: backend/orchestrator-prompt/system/design_document_review.py
# 設計書のレビュー
def get_system_message(
    design_document_list: str,
    reviewPurpose: str,
    priorityPoint: str,
    consideration: str,
) -> str:
    DESIGN_DOCUMENT_REVIEW_SYSTEM_MESSAGE = f"""\
    # 役割
    あなたは優秀な設計者であり、プロの設計レビュアーです。
    読者に配慮した、丁寧で建設的、かつ具体的なフィードバックと提案を行います。

    # 目的
    新製品の設計書をレビューし、改善点と実行可能な提案をわかりやすく提示します。

    # 指示
    以下の設計書をレビューし、丁寧で具体的な「フィードバック」と「提案」のみを出力してください。
    断定口調は避け、根拠と理由を添え、必要に応じて代替案・改善手順・記載例を示してください。

    # 入力欄
    設計書の内容：{design_document_list}
    レビューの目的：{reviewPurpose}
    特に見てほしい箇所：{priorityPoint}
    考慮事項：{consideration}

    # 制約条件
    - 前置き・要約・謝辞は不要。フィードバックと提案のみを記述すること。
    - すべての指摘には、対応する提案（改善案・代替手段・具体例・次アクション）を併記すること。
    - 指摘箇所は、可能な限り「場所」を明記すること（例：p.12, 図3-2, 章2.4「API仕様」など）。\
        該当箇所が特定できない場合は「全体方針」セクションでまとめること。\
            ページ番号や見出しが不明な場合は、本文の引用またはキーワードを「場所」に記載する（例：「場所: “スループットは十分”の段落」）
    - {reviewPurpose},{priorityPoint},{consideration}が空の場合は、\
      {design_document_list}のみをもとにレビューすること。
    - {reviewPurpose},{priorityPoint},{consideration}がある場合は、\
      それに基づいてフィードバックを提供すること。
    - {priorityPoint} がある場合は、その領域を最初にセクション分けして集約し、その後にその他の指摘を記載すること。
    - 文章は丁寧で読みやすく、箇条書きと短い段落を組み合わせて表現すること。

    # 出力形式
    1. マークダウン形式で出力。以下の章立てを基本とする。
    - セクション: 「特に見てほしい箇所（{priorityPoint}）」［ある場合のみ］
    - セクション: 「個別フィードバック（場所別）」\
        小見出し単位で「場所」「フィードバック」「提案」をセットで記載
    - セクション: 「全体方針・横断的な改善」
    例：個別フィードバックの書式
    - 場所: [具体的な場所、ページ数や見出し名を記入]
    - フィードバック: [具体的なフィードバックを記入]
    - 提案: [具体的な提案を記入]

    """
    return DESIGN_DOCUMENT_REVIEW_SYSTEM_MESSAGE


def get_user_message(
    design_document_list: str,
    reviewPurpose: str,
    priorityPoint: str,
    consideration: str,
):
    messages = [
        {
            "role": "user",
            "content": get_system_message(
                design_document_list,
                reviewPurpose,
                priorityPoint,
                consideration,
            ),
        },
    ]
    return messages

File: frontend/src/app/(dashboard)/design-document-review/_actions/designDocumentReview.ts
'use server';

import { AzureResponse, DesignDocumentReviewResponse } from '@/app/_actions/types';
import { designDocumentReviewDB } from '@/app/_db/design-document-review';
import { LLMserviceBackEndLog } from '@/app/_types/logs/llm-services';
import { getCurrentUser } from '@/app/_utils/auth';
import { getMessage } from '@/app/_utils/message';
import { useCaseAzureFunctions } from '../../../../../azure-functions';
import { designDocumentReviewContainer } from '../../../../../cosmos';

export async function designDocumentReview(
  id: string,
  formData: FormData
): Promise<DesignDocumentReviewResponse> {
  const fileList = formData.get('fileList') ? (formData.getAll('fileList') as File[]) : [];
  const user = await getCurrentUser();

  if (fileList.length === 0) {
    throw new Error('ファイルを１つ以上選択してください');
  }

  try {
    const answerResponse = await useCaseAzureFunctions.sendForm<
      AzureResponse & {
        temp_file: string;
        log: LLMserviceBackEndLog<'design-document-review'>;
      }
    >('design-document-review', formData);

    // 失敗時
    if (!answerResponse.success) {
      const errorAnswerResponse = answerResponse.message;
      console.error('メッセージの送信が失敗しました。', errorAnswerResponse);
      throw new Error(errorAnswerResponse);
    }
    // log
    await designDocumentReviewDB(designDocumentReviewContainer, {
      id,
      userId: user.id,
      useName: user.name,
      userEmail: user.email,
      userDepartmentName: user.departmentName,
      createdAt: new Date(),
      reviewPurpose: formData.get('reviewPurpose')?.toString() || '',
      priorityPoint: formData.get('priorityPoint')?.toString() || '',
      consideration: formData.get('consideration')?.toString() || '',
      outputForm: answerResponse.answer,
      inputForm: answerResponse.temp_file,
      log: answerResponse.log,
    });

    return {
      success: true,
      content: answerResponse.answer,
    };
  } catch (error) {
    console.error('Error design-document-review:', error);
    return {
      message: error instanceof Error ? error.message : getMessage('E_F_00110', '作成結果'),
      success: false,
    };
  }
}
